main.go

```diff
@@ -231,32 +195,45 @@ func (httpConf httpConfigurations) readMessage(msg queueConsumer.Message) {
 			uuid = v
 		}
 	}
-	reqURL, err := sendToWriter(ingestionType, strings.NewReader(msg.Body), uuid, httpConf.baseURLMap)
+	reqURL, err := sendToWriter(ingestionType, strings.NewReader(msg.Body), uuid, httpConf.baseURLSlice)
 
 	if err != nil {
 		log.Errorf("Error processing msg: %v with error %v to %v", msg, err, reqURL)
 	}
 }
 
-func sendToWriter(ingestionType string, msgBody *strings.Reader, uuid string, urlMap map[string]string) (reqURL string, err error) {
-	writerURL := urlMap[ingestionType]
+func sendToWriter(ingestionType string, msgBody io.Reader, uuid string, URLSlice []string) (reqURL string, err error) {
```

Comment:
Any reason for returning reqURL? It is only used for logging purposes. Maybe create a new error containing the proper err message in this function and return the error only?